### PR TITLE
Feature/add docz

### PR DESCRIPTION
**Pull request by Grsmto**
- Add Docz to the project:
  How to test:
  - `yarn`
  - `yarn docz:dev`

- Added a build step (open source JS code should be transpiled for people to use)
- Refactored Jest setup

Know issues: at the moment the Docz build breaks when importing anything from the `polymath-ui` package (with a missing dependencies error). So every components that import it are commented.

### CODE REVIEW

File: packages/polymath-ui/doczrc.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { css } from 'docz-plugin-css';
+
+export default {
+  title: 'PolymathUI',
+  description: 'This is my awesome documentation',
+  files: '**/*.{md,mdx}',
+  // TODO @grsmto: we can't include this wrapper before it imports some polymath-js dependencies that break Docz build. Need to investigate why this is happening.
```
Comment (RafaelVidaurre) [inline packages/polymath-ui/doczrc.js]: What error are you getting. PolymathJS right now has a big issue where it **requires** `Contract.setParams` to be called with network parameters before anything inside it is used

Comment (Grsmto) [reply packages/polymath-ui/doczrc.js]:
```
These dependencies were not found:

* @polymathnetwork/shared/fixtures/contracts/CappedSTOFactory.json in ../polymath-js/src/contracts/CappedSTOFactory.js
* @polymathnetwork/shared/fixtures/contracts/CountTransferManagerFactory.json in ../polymath-js/src/contracts/CountTransferManage
rFactory.js
* @polymathnetwork/shared/fixtures/contracts/PercentageTransferManagerFactory.json in ../polymath-js/src/contracts/PercentageTran
sferManagerFactory.js

To install them, you can run: npm install --save @polymathnetwork/shared/fixtures/contracts/CappedSTOFactory.json @polymathnetwor
k/shared/fixtures/contracts/CountTransferManagerFactory.json @polymathnetwork/shared/fixtures/contracts/PercentageTransferManager
Factory.json
```

**Comment by Grsmto** [comment]
I edited the PR afterward, dunno if you saw:
`Added a build step (open source JS code should be transpiled for people to use)`

It's not the build script from CRA, I just took it as boilerplate. This is because I needed it to be within a JS file to be able to put extra stuff like `process.env.NODE_ENV=production` and not have it inlined in the package.json. I can remove the extra junk if you want tho!


Also I moved the Webpack config to get something consistent across packages: investors and issuers have all the configs in a /config folder (I'm used to that kind of architecture as well). So I just did the same here.

File: packages/polymath-ui/doczrc.js (hunk 1 above)
Comment (RafaelVidaurre) [reply packages/polymath-ui/doczrc.js]: Oh, I guess it could be the babel config not having moduleResolver?

**Comment by Grsmto** [comment]
Ah yeah damn I reset some files by mistake. Should be working fine now.

**Pull request merged**